### PR TITLE
Make Helm release garbage collection interval configurable

### PR DESCRIPTION
**Pull request by weyfonk**
Refers to #2169.

This also adds end-to-end tests validating that old releases are cleaned up when updating a bundle's namespace or release name.

### CODE REVIEW

File: e2e/assets/helm/repo/with-release-name/crds-only/crds-chart/templates/crds.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            foo:
+              description: 'Foo is important.'
+              nullable: true
+              type: string
+            bar:
+              description: 'Bar does not walk into a Foo. A Foo walks into a bar.'
+              nullable: true
+              type: string
+            kind:
+              description: 'Kindness is everything.'
```
Comment (0xavi0) [inline e2e/assets/helm/repo/with-release-name/crds-only/crds-chart/templates/crds.yaml:33]: :heart: the descriptions

File: internal/cmd/controller/agentmanagement/controllers/cluster/import.go [hunk 2]
```diff
@@ -97,7 +97,10 @@ func (i *importHandler) onConfig(config *config.Config) error {
 
 		hasConfigChanged := config.APIServerURL != cluster.Status.APIServerURL ||
 			hashStatusField(config.APIServerCA) != cluster.Status.APIServerCAHash ||
-			config.AgentTLSMode != cluster.Status.AgentTLSMode
+			config.AgentTLSMode != cluster.Status.AgentTLSMode ||
+			(config.GarbageCollectionInterval.Duration != 0 && cluster.Status.GarbageCollectionInterval == nil) ||
+			(cluster.Status.GarbageCollectionInterval != nil &&
+				config.GarbageCollectionInterval.Duration != cluster.Status.GarbageCollectionInterval.Duration)
```
Comment (0xavi0) [inline internal/cmd/controller/agentmanagement/controllers/cluster/import.go:103]: nit: maybe create a helper function that do those checks about `GarbageCollectionInterval`.
I found it hard to read as it is a pretty large expression, the other checks are pretty simple in comparison.

**Review by 0xavi0 — APPROVED** [review]
lgtm.
Just a nit about a expression that I found too large, but not enough to not approve this.
Nice one

File: internal/cmd/controller/agentmanagement/controllers/cluster/import.go (hunk 2 above)
Comment (weyfonk) [reply internal/cmd/controller/agentmanagement/controllers/cluster/import.go]: Good point, thanks for raising this. I've moved this to a separate function.

**Review by 0xavi0 — APPROVED** [review]

**Pull request merged**